src/praisonai-bot/praisonai_bot/gateway/kanban_dispatcher.py

```
@@ -481,14 +481,61 @@ def _prepare_worktree(self, task: Any, store: Any) -> Optional[str]:
+        if not force and branch is not None:
+            reason = self._worktree_outstanding_work(path, branch)
+            if reason is not None:
+                logger.warning(
+                    "Preserving worktree %s (not removing): %s", path, reason,
+                )
+                return reason
         try:
             result = self._run_git("worktree", "remove", "--force", path)
             if result.returncode != 0:
                 logger.debug(f"worktree remove failed for {path}: {result.stderr.strip()}")
         except Exception as e:
```

Ruff (0.15.21)

[warning] Do not catch blind exception: `Exception`

(BLE001)